ts/src/base/functions/customThrottle.ts

```diff
@@ -20,23 +25,17 @@ class Throttler {
             },
         };
         Object.assign (this.config, config);
-        this.queue = [];
-        this.priorityQueue = {};
+        this.priorityQueue = {}; // Set priority here
         this.running = false;
     }
 
     /**
      * Returns the most important queue (lower number => hight priority), if there are no priorities, returns the normal queue
      */
     getMostImportantQueue () {
-        const prioritieIndices = Object.keys (this.priorityQueue);
-        let highestPriorityIndex = 0;
-        let highestPriorityQueue = this.queue;
-
-        if (prioritieIndices.length !== 0) {
-            highestPriorityIndex = Math.min (...prioritieIndices);
-            highestPriorityQueue = this.priorityQueue[highestPriorityIndex];
-        }
+        const priorityIndices = Object.keys (this.priorityQueue);
+        const highestPriorityIndex = Math.min (...priorityIndices);
+        const highestPriorityQueue = this.priorityQueue[highestPriorityIndex];
```

Comment:
By default, the queue is `{}`. In this case `Math.min (...priorityIndices);` will be equal to `0`, and `this.priorityQueue[highestPriorityIndex];` will potentially return `undefined`, right? 

Then we call `getMostImportantQueue` in the `loop` and get the first element from `undefined` by doing `highestPriorityQueue[0];` (line 51). In that case it will throw an error `can't read from undefined`. Did you cover this case?